src/utils/finance/source.py

```diff
@@ -294,25 +326,74 @@ def _standardize_null_values(
     ) -> pd.DataFrame:
         """Replace implicit null values with actual null values"""
         for column_name, null_values in self.null_values.items():
+            if column_name not in standard_schema_table.columns:
+                continue
             for implicit_null_value in null_values:
                 standard_schema_table[column_name] = standard_schema_table[
                     column_name
                 ].replace(implicit_null_value, pd.NA)
         return standard_schema_table
 
-    def _filter_data(self, standard_schema_table: pd.DataFrame) -> pd.DataFrame:
-        """Filter data based on filter configuration"""
-        for column_name, filter_values in self.filter.items():
-            if column_name not in standard_schema_table.columns:
-                continue
-            if "NOT" in filter_values:
-                standard_schema_table = standard_schema_table[
-                    ~standard_schema_table[column_name].isin(filter_values["NOT"])
-                ]
+    def _standardize_transaction_direction(
+        self, standard_schema_table: pd.DataFrame
+    ) -> pd.DataFrame:
+        """Flip donor and recipient columns if 'transaction_direction' is 'reverse'
+
+        In state config, if a raw column is mapped to transaction_direction and
+        the enum mapper maps some values to 'reverse' those rows will switch
+        donor and recipient columns.
+        """
+        if "transaction_direction" not in standard_schema_table.columns:
+            return standard_schema_table
+        reverse_mask = standard_schema_table["transaction_direction"] == "reverse"
+        reverse_table = standard_schema_table.loc[reverse_mask, :]
```

Comment:
It makes me nervous that you're not copying the table, but if you're sure this works I trust you. When you assign a value reverse_table.columns, that doesn't also reassign the columns for the original table? I suppose I wouldn't expect it too, but I would have to test it to be confident.